src/material/core/option/index.ts

```diff
@@ -9,13 +9,19 @@
 import {NgModule} from '@angular/core';
 import {CommonModule} from '@angular/common';
 import {MatRippleModule} from '../ripple/index';
-import {MatPseudoCheckboxModule} from '../selection/index';
+import {MatPseudoCheckboxModule, MatPseudoCheckmarkModule} from '../selection/index';
 import {MatCommonModule} from '../common-behaviors/common-module';
 import {MatOption} from './option';
 import {MatOptgroup} from './optgroup';
 
 @NgModule({
-  imports: [MatRippleModule, CommonModule, MatCommonModule, MatPseudoCheckboxModule],
+  imports: [
+    MatRippleModule,
+    CommonModule,
+    MatCommonModule,
+    MatPseudoCheckboxModule,
+    MatPseudoCheckmarkModule,
```

Comment:
I don't think that we should create an entirely new component for this. Maybe we can reuse the existing pseudo checkbox and add an `appearance` input to it? E.g. something like `appearance="minimal` would remove the checkbox rectangle and invert the color of the checkmark.

Reply:
Okay, like what I did in #25870, would that work? I made the appearance options `'checkmark'` or `'checkbox'`. The checkmark looked okay, but it might look better if the checkmark were a bit bigger since it doesn't have the box around it.

```
@Input() appearance: 'checkmark' | 'checkbox' = 'checkbox';
```

Reply:
^ Yeah, I had that same idea after doing https://github.com/angular/components/pull/25870. That's why I that it would make more sense to have a different component for checkmark, rather than a checkbox with an appearance option.

```
<!-- Option B: separate component for checkmark -->
<mat-pseudo-checkbox></mat-pseudo-checkbox>
<mat-pseudo-checkmark></mat-pseudo-checkmark>
```

As a third option there is also rename mat-pseudo-checkbox to mat-pseudo-toggle and give appearance for checkbox/checkmark
```
<!-- Option C: rename mat-pseudo-checkbox to mat-pseudo-toggle-->
<mat-pseudo-toggle appearance="checkbox"></mat-pseudo-toggle>
<mat-pseudo-toggle appearance="checkmark"></mat-pseudo-toggle>
```

I'm leaning towards options B or C. I think having a component named checkbox but actually giving you checkmark is confusing. Checkmark vs checkbox mean very different things because one signals single-select and the other signals multi-select.

Reply:
Well, checkmark can be used for both single-select and multi-selection. For example, the chips use checkmark for multi-select. the point is that checkbox and checkmark mean different things semantically.

Reply:
It's the same component though, it just looks differently. The separate component will make it more complicated to switch between the two, because we'd have to wrap it in `ngIf` or `ngSwitch`. It's only a matter of what to call the apperances.

Reply:
Okay, that's fair. I'm pondering which option will meet our needs and also be the least opinionated.

how do you feel about Option C? It's the same component, and it's also clear to the developer what visual they will get based on the API. I believe mat-pseudo-checkbox is not public API, so we can rename it (need to double check that).

Reply:
Option C feel weird to me since we're basically only showing some variation of a checkmark, with or without a box around it. "Toggle" also makes me think of something like `mat-slide-toggle`.

Reply:
I think toggle is a really conventional way to describe something that has a boolean state. Although, as maintain of mdc-list, I think you have the most say in this :).

What's your preferred API?

Is it
```
@Input() appearance: 'minimal' | 'full' = 'full'; // render checkbox by default. 'minimal' renders a checkmark
```

Also does adding `@docs-private` make `MatPseudoCheckbox` private API, or just hide the comment from the docs?

Reply:
That API LGTM. `@docs-private` only hides it from the docs site, but it won't prevent people from using it if they import the module.

Reply:
Thanks, I'll go ahead and implement the appearance minimal/full API with full as the default. That API doesn't look great to me. From an angular developers point of view, I would imagine it would be confusing when shopping for a component to use. That's because when I see the word "checkbox" I would expect it to only be used for multi-selection.

minimal/full is okay, and I'm willing to do that so that we can land this a11y improvement. I'll go ahead and close this PR and send out a new PR for review when I have it working.

Reply:
I implemented the 'minimal'|'full' apperance as suggested, and I think you were right about it. The API made a lot more sense to me after I looked at all the pseudo checkboxes together on the checkbox demo.

Reply:
https://github.com/angular/components/pull/25962/